core/base/src/TApplication.cxx

```diff
@@ -628,20 +630,453 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
+   // extended ".help" command or the old one.
+   if (input[0] != nullptr) {
+      TString className(input[0]);
+      // We check if "className" exists.
+      if ((TClass::GetClass(className)) != nullptr) {
+          TClass *classVariable = TClass::GetClass(className);
+         // We check if "classVariable" isn't a namespace.
+         if ((classVariable->Property() & kIsNamespace) == 0) {
+            // We check if the user wants to search for a certain method from the class.
+            if (input[1] != nullptr) {
+               // We take the name of the function (the text before the first bracket).
+               // In case the user uses "Tab" to finish the name of the function, he will not
+               // have to delete by hand the sumbol "(".
+               // This also means that ".help" for "operator()" won't work.
+               TString memberFunc(strtok(input[1], "("));
+               TString tilde("~");
+               // We check if "memberFunc" is an enumerator.
+               if ((classVariable-> GetListOfEnums()->FindObject(memberFunc))!=nullptr){
+                  // The URL consists of URL for the "className", "#a" and crypted with MD5 text.
+                  // The text is "Class::EnumeratorEnumerator.
+                  TString md5Enumerator(className);
+                  md5Enumerator.Append("::");
+                  md5Enumerator.Append(memberFunc);
+                  md5Enumerator.Append(memberFunc);
+                  // We make the URL for the class "className".
+                  TString url(UrlGenerator(className, kURLforClass));
+                  // Then we have to append "#a" and the crypted text.
+                  url.Append("#a");
+                  url.Append(md5Enumerator.MD5());
+                  // We call "OpenInBrowser" to load the correct URL in web browser.
+                  OpenInBrowser(url);
+               } else {
+                  // We check if "memberFunc" is a part of enumerator defined inside "className" or its base classes.
+                  if (((classVariable->GetListOfAllPublicDataMembers()->FindObject(memberFunc)) != nullptr) ||
+                  ((classVariable->GetListOfDataMembers()->FindObject(memberFunc)) != nullptr)) {
+                     // If so we check if it is defined inside "className".
+                     if ((classVariable->GetListOfDataMembers()->FindObject(memberFunc)) != nullptr) {
+                        // We generate the URL with "URLforEnum"
+                        OpenInBrowser(URLforEnum(className, classVariable, memberFunc));
+                     } else {
+                        // If it's not part of "className" definition, then it is defined in one of its base classes.
+                        // We find out the name of it.
+                        TString baseClass(((TMethod *)(((TDataMember *)classVariable->GetListOfAllPublicDataMembers()->FindObject(memberFunc)))->GetClass())->GetName());
```

Comment:
Same here, these are expensive operations, please use the result of the previous `FindObject` call.